### PR TITLE
Disable kSendTabToSelfManageDevicesLink also for desktop (uplift to 1.36.x)

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Uplift of #12244
Resolves https://github.com/brave/brave-browser/issues/21082

Pre-approval checklist: 
- [x] You have tested your change on Nightly. 
- [ ] This contains text which needs to be translated. 
    - [ ] There are more than 7 days before the release. 
    - [ ] I've notified folks in #l10n on Slack that translations are needed. 
- [x] The PR milestones match the branch they are landing to. 

Pre-merge checklist: 
- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 

Post-merge checklist: 
- [x] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Comment by AlexeyBarabash** [comment]
macOS build is marked unstable, not related to the PR: 
https://ci.brave.com/job/pr-brave-browser-pr12244_fix_21082_1.36.x-macos/1/execution/node/273/log/
```
14:37:13  [51882:259:0216/043712.418847:WARNING:render_frame_host_impl.cc(1202)] InterfaceRequest was dropped, the document is no longer active: blink.mojom.CodeCacheHost
14:37:13  ../../brave/browser/extensions/api/brave_shields_api_browsertest.cc:208: Failure
14:37:13  Expected equality of these values:
14:37:13    GetActiveRenderFrameHosts().size()
14:37:13      Which is: 3
14:37:13    2u
14:37:13      Which is: 2
14:37:13  All script loadings should be blocked.
14:37:13  Stack trace:
14:37:13  0   brave_browser_tests                 0x0000000105c76c5b extensions::BraveShieldsAPIBrowserTest_AllowScriptsOnceIframe_Test::RunTestOnMainThread() + 907
14:37:13  1   brave_browser_tests                 0x000000010fb433f8 content::BrowserTestBase::ProxyRunTestOnMainThreadLoop() + 984
14:37:13  2   brave_browser_tests                 0x0000000107c94c20 content::BrowserMainLoop::RunMainMessageLoop() + 64
14:37:13  3   brave_browser_tests                 0x0000000107c96732 content::BrowserMainRunnerImpl::Run() + 82
14:37:13  4   brave_browser_tests                 0x0000000107c920fd content::BrowserMain(content::MainFunctionParams) + 253
14:37:13  5   brave_browser_tests                 0x0000000109400344 content::RunBrowserProcessMain(content::MainFunctionParams, content::ContentMainDelegate*) + 260
14:37:13  6   brave_browser_tests                 0x0000000109401813 content::ContentMainRunnerImpl::RunBrowser(content::MainFunctionParams, bool) + 1235
14:37:13  7   brave_browser_tests                 0x000000010940125e content::ContentMainRunnerImpl::Run() + 558
14:37:13  8   brave_browser_tests                 0x00000001093ff39d content::RunContentProcess(content::ContentMainParams, content::ContentMainRunner*) + 621
14:37:13  9   brave_browser_tests                 0x00000001093ffdd7 content::ContentMain(content::ContentMainParams) + 71
14:37:13  10  brave_browser_tests                 0x000000010fb423ef content::BrowserTestBase::SetUp() + 2095
14:37:13  11  brave_browser_tests                 0x000000010c03efee InProcessBrowserTest::SetUp() + 446
14:37:13  
14:37:13  [51882:47363:0216/043713.130693:ERROR:process_posix.cc(326)] Unable to terminate process 51891: No such process (3)
14:37:13  [  FAILED  ] BraveShieldsAPIBrowserTest.AllowScriptsOnceIframe, where TypeParam =  and GetParam() =  (2243 ms)
```

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.36.x` approved after deliberating with @brave/uplift-approvers. QA has verified the PR on Nightly as per https://github.com/brave/brave-core/pull/12244#issuecomment-1042149839.

**Pull request merged**
